File: backend/src/main/java/me/performancereservation/global/storage/redis/RedisSeatInitializationListener.java
package me.performancereservation.global.storage.redis;

import lombok.RequiredArgsConstructor;
import me.performancereservation.domain.performance.event.PerformanceScheduleCreatedEvent;
import org.springframework.context.event.EventListener;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class RedisSeatInitializationListener {
    private final RedisSeatService redisSeatService;

    @EventListener
    public void handlePerformanceScheduleCreated(PerformanceScheduleCreatedEvent event) {
        redisSeatService.initializeSeatStock(event.scheduleId(), event.totalSeats());
    }
}

File: backend/src/main/kotlin/me/performancereservation/global/storage/redis/RedisSeatInitializationListener.kt
package me.performancereservation.global.storage.redis

import me.performancereservation.domain.performance.event.PerformanceScheduleCreatedEvent
import org.springframework.context.event.EventListener
import org.springframework.stereotype.Component

@Component
class RedisSeatInitializationListener(
    private val redisSeatService: RedisSeatService
) {
    @EventListener
    fun handlePerformanceScheduleCreated(event: PerformanceScheduleCreatedEvent) {
        redisSeatService.initializeSeatStock(
            scheduleId = event.scheduleId,
            stock = event.totalSeats
        )
    }
}
